lib/index.js

```diff
@@ -15,329 +15,303 @@ const internals = {};
 
 exports.genMigrationFile = (options, cb) => {
 
-    Joi.assert(options, Schema.options, 'Bad options passed to schwifty-migration.');
+    const validOptions = Joi.validate(options, Schema.options);
+    if (validOptions.error) {
+        return cb(new Error(`Bad options passed to schwifty-migration: ${validOptions.error.message}`));
```

Comment:
When a callback is called synchronously it should instead by wrapped in `process.nextTick()` to avoid oddities.

Reply:
Oh interesting! What kind of oddities might happen?